server/channels/app/security_update_check.go

```diff
@@ -64,9 +64,15 @@ func (s *Server) DoSecurityUpdateCheck() {
 
 		systemSecurityLastTime := &model.System{Name: model.SystemLastSecurityTime, Value: strconv.FormatInt(currentTime, 10)}
 		if lastSecurityTime == 0 {
-			s.Store().System().Save(systemSecurityLastTime)
+			if err := s.Store().System().Save(systemSecurityLastTime); err != nil {
+				s.Log().Error("Failed to save last security check time", mlog.Err(err))
+				return
+			}
 		} else {
-			s.Store().System().Update(systemSecurityLastTime)
+			if err := s.Store().System().Update(systemSecurityLastTime); err != nil {
+				s.Log().Error("Failed to update last security check time", mlog.Err(err))
+				return
+			}
```

Comment:
@hanzei do we want to return early here on error? Not sure if saving the last time should be a reason to not continue with the check and send the security bulletins email

Reply:
I won't be on a loop. If it fails to save it will either: not have a value or have the old timestamp and execute it again in the next run 4 hours later. This could lead to a double email sent so I'm good if we return here on a fail as it will try again later.

Reply:
1/5 I'm leaning towards failing to avoid sending emails too often.